shared/src/utils/validators/is-string-literal-union.ts

```diff
@@ -0,0 +1,26 @@
+import type { ValidationOptions, ValidationArguments } from 'class-validator';
+import { StrictObject } from '../strict-object';
+import { makeValidator } from './make-validator';
+
+export function isStringLiteralUnion<T extends string>(
+    allowedValues: { [key in T]: true },
+    valueToBeValidated: unknown
+): boolean {
+    return (
+        typeof valueToBeValidated === 'string' &&
+        StrictObject.keys(allowedValues).includes(valueToBeValidated as T)
```

Comment:
```suggestion
        allowedValues[valueToBeValidated as T]
```

We already have a Set ^^

Reply:
Good point. Will change that.